vm_eval.c

```diff
@@ -163,6 +163,19 @@ vm_call0_cfunc(rb_execution_context_t *ec, struct rb_calling_info *calling, cons
     return vm_call0_cfunc_with_frame(ec, calling, argv);
 }
 
+static void
+vm_call_check_arity(struct rb_calling_info *calling, int argc, const VALUE *argv)
+{
+    if (calling->kw_splat &&
+        calling->argc > 0 &&
+        RB_TYPE_P(argv[calling->argc-1], T_HASH) &&
+        RHASH_EMPTY_P(argv[calling->argc-1])) {
+        calling->argc--;
+    }
```

Comment:
It could be good to comment this to explain what this check is testing

Reply:
@jeremyevans may know.

Reply:
This is testing for empty splatted keyword hashes.  If an empty splatted keyword hash are passed, it should be equivalent to not passing keywords at all.